serde/src/private/de.rs

```diff
@@ -1715,16 +1717,10 @@ mod content {
         V: Visitor<'de>,
         E: de::Error,
     {
-        fn content_ref_deserializer_pair<'a, 'de, E>(
+        fn content_ref_deserializer_pair<'a, 'de>(
```

Comment:
I also tried adding:

```rust
impl<'a, A, B> Pair for &'a (A, B) {
    type First = &'a A;
    type Second = &'a B;
    fn split(self) -> (&'a A, &'a B) {
        (&self.0, &self.1)
    }
}
```

to make this function go away, like in the `visit_content_map` case. See #2823.

But somehow that made the rlib go back up from 111K to 124K, negating the benefit from #2821. I do not fully understand why this happens.